data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/sink/FileSink.java

```diff
@@ -61,23 +63,33 @@ public FileSink(final FileSinkConfig fileSinkConfig, final SinkContext sinkConte
         initialized = false;
         lock = new ReentrantLock(true);
         tagsTargetKey = Objects.nonNull(sinkContext) ? sinkContext.getTagsTargetKey() : null;
+        this.sinkContext = sinkContext;
     }
 
     @Override
     public void output(final Collection<Record<Object>> records) {
+        final boolean doForward = sinkContext.getForwardToPipelines().size() > 0;
         lock.lock();
+        Collection<Record<Event>> events = new ArrayList<>();
         try {
             if (isStopRequested)
                 return;
 
             for (final Record<Object> record : records) {
                 try {
                     checkTypeAndWriteObject(record.getData(), writer);
+                    if (doForward && record.getData() instanceof Event) {
+                        Event event = (Event)record.getData();
+                        events.add(new Record<>(event));
+                    }
                 } catch (final IOException ex) {
                     throw new RuntimeException(format("Encountered exception writing to file %s", outputFilePath), ex);
                 }
             }
 
+            if (doForward) {
```

Comment:
I think it would be nice to call the `SinkContext` to get a "forwardTo" batch. Then you populate it and forward it. This can move these conditions in there so we don't repeat a lot of possibly fragile code.

```
ForwardBatch forwardBatch = sinkContext.getForwardBatch();
...

checkTypeAndWriteObject(record.getData(), writer);
forwardBatch(record);
...

forwardBatch.forward();
```